crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! AnyValue specification.
+
+use std::fmt::{Display, Formatter};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::attribute::Examples;
+use crate::stability::Stability;
+
+/// The AnyValueTypeSpec is a specification of a value that can be of any type.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum AnyValueSpec {
```

Comment:
This structure is mostly unchanged from the file @lquerel provided back (added comments, and some additional tests).

Creating this Draft PR so that we can have an isolated discussion about the "AnyValue" support only.

I'm going to start merging this (and any updates we discuss) into #297